core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3624,7 +3624,9 @@ public FileSystemCommand workerHeartbeat(long workerId, List<Long> persistedFile
     TtlAction action = options.getTtlAction();
 
     if (ttl != null && inode.getTtl() != ttl) {
+      mTtlBuckets.remove(inode);
       inode.setTtl(ttl);
+      mTtlBuckets.insert(inode);
```

Comment:
This touches the DefaultFileSystemMaster and not a test. Is this something that should also go into master?